core.impl/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/impl/table/TableStorageStatesImpl.java

```diff
@@ -0,0 +1,192 @@
+                builder.myInitialDelayInMs,
+                builder.myUpdateDelayInMs,
+                TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    public long getDataSize(final UUID nodeID, final TableReference tableReference)
+    {
+        Map<UUID, ImmutableMap<TableReference, Long>> dataSizes = myTableSizes.get();
+
+        if (dataSizes != null && dataSizes.containsKey(nodeID) && dataSizes.get(nodeID).containsKey(tableReference))
+        {
+                return dataSizes.get(nodeID).get(tableReference);
+        }
+        return 0;
+    }
+
+    @Override
+    public long getDataSize(final UUID nodeID)
+    {
+        Map<UUID, ImmutableMap<TableReference, Long>> dataSizesMap = myTableSizes.get();
+        ImmutableMap<TableReference, Long> dataSizes = dataSizesMap.get(nodeID);
+
+        if (dataSizes != null)
+        {
+            return dataSizes.values().stream().mapToLong(e -> e).sum();
+        }
+
+        return 0;
+    }
+
+    @Override
+    public void close()
+    {
+        myScheduledExecutorService.shutdown();
+
+        myTableSizes.set(null);
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
+    {
+        private ReplicatedTableProvider myReplicatedTableProvider;
+        private DistributedJmxProxyFactory myJmxProxyFactory;
+
+        private long myInitialDelayInMs = 0;
+        private long myUpdateDelayInMs = DEFAULT_UPDATE_DELAY_IN_MS;
+
+        public final Builder withReplicatedTableProvider(final ReplicatedTableProvider replicatedTableProvider)
+        {
+            myReplicatedTableProvider = replicatedTableProvider;
+            return this;
+        }
+
+        public final Builder withJmxProxyFactory(final DistributedJmxProxyFactory jmxProxyFactory)
+        {
+            myJmxProxyFactory = jmxProxyFactory;
+            return this;
+        }
+
+        public final Builder withInitialDelay(final long initialDelay, final TimeUnit timeUnit)
+        {
+            myInitialDelayInMs = timeUnit.toMillis(initialDelay);
+            return this;
+        }
+
+        public final Builder withUpdateDelay(final long updateDelay, final TimeUnit timeUnit)
+        {
+            myUpdateDelayInMs = timeUnit.toMillis(updateDelay);
+            return this;
+        }
+
+        public final TableStorageStatesImpl build()
+        {
+            if (myReplicatedTableProvider == null)
+            {
+                throw new IllegalArgumentException("Replicated table provider cannot be null");
+            }
+
+            if (myJmxProxyFactory == null)
+            {
+                throw new IllegalArgumentException("JMX proxy factory cannot be null");
+            }
+
+            return new TableStorageStatesImpl(this);
+        }
+    }
+
+    @VisibleForTesting
+    void updateTableStates()
+    {
+        if (myJmxProxyFactory != null)
+        {
+            for (UUID nodeID : myTableSizes.get().keySet())
```

Comment:
MyTableSizes is empty when it first enters this method, so no new table sizes are calculated, and MyTableSizes stays empty, meaning the sizes are never calculated

Reply:
Nice find, I implemented now a method to initialize the table stats map with the nodes list